src/Fixer/Import/NoUnusedImportsFixer.php

```diff
@@ -105,26 +106,45 @@ private function isImportUsed(Tokens $tokens, NamespaceAnalysis $namespace, Name
 
         $tokensNotBeforeFunctionCall = [T_NEW];
 
-        if (\defined('T_ATTRIBUTE')) { // @TODO: drop condition when PHP 8.0+ is required
+        $attributeIsDefined = \defined('T_ATTRIBUTE');
+
+        if ($attributeIsDefined) { // @TODO: drop condition when PHP 8.0+ is required
             $tokensNotBeforeFunctionCall[] = T_ATTRIBUTE;
         }
 
         $namespaceEndIndex = $namespace->getScopeEndIndex();
+        $inAttribute = false;
 
         for ($index = $namespace->getScopeStartIndex(); $index <= $namespaceEndIndex; ++$index) {
+            $token = $tokens[$index];
+
+            if ($attributeIsDefined && $token->isGivenKind(T_ATTRIBUTE)) {
+                $inAttribute = true;
+
+                continue;
+            }
+
+            if ($attributeIsDefined && $token->isGivenKind(CT::T_ATTRIBUTE_CLOSE)) {
+                $inAttribute = false;
+
+                continue;
+            }
+
             if (isset($ignoredIndexes[$index])) {
                 $index = $ignoredIndexes[$index];
 
                 continue;
             }
 
-            $token = $tokens[$index];
-
             if ($token->isGivenKind(T_STRING)) {
                 if (0 !== strcasecmp($import->getShortName(), $token->getContent())) {
                     continue;
                 }
 
+                if ($inAttribute) {
+                    return true;
```

Comment:
maybe this should be `return $import->getType() === NamespaceUseAnalysis::TYPE_CLASS;` or can you use function and const references in attributes as well?

Reply:
Luckily no, only classes can be attributes.